File: app/src/main/java/com/frogobox/evpn/BaseApplication.java
package com.frogobox.evpn;

import android.app.Application;
import android.content.Context;

public class BaseApplication extends Application {

    private static BaseApplication instance;

    public static String getResourceString(int resId) {
        return instance.getString(resId);
    }

    public static BaseApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
    }

}

File: app/src/main/java/com/frogobox/viprox/BaseApplication.kt
package com.frogobox.viprox

import android.app.Application
import android.content.Context

class BaseApplication : Application() {

    companion object {
        lateinit var instance: BaseApplication
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
    }

    fun getResourceString(resId: Int): String {
        return instance.getString(resId)
    }

}